spec-tests/src/test/java/org/candlepin/spec/ExportSpecTest.java

```diff
@@ -0,0 +1,329 @@
+        // All 6 should be there, despite one cert is for a virt product
+        assertEquals(6, entitlementCerts.length);
+
+        for (File entitlementCert : entitlementCerts) {
+            String cert = Files.readString(entitlementCert.toPath());
+            assertEquals("-----BEGIN CERTIFICATE-----", cert.substring(0, 27));
+        }
+    }
+
+    @Test
+    @DisplayName("should export rules")
+    void shouldExportRules() throws Exception {
+        List<File> rules2Dirs = FileUtil.getSubdirectories("rules2", export);
+        assertEquals(1, rules2Dirs.size());
+
+        String expectedRules = rulesApi.getRules();
+        Base64 base64 = new Base64();
+        expectedRules = new String(base64.decode(expectedRules.getBytes()));
+
+        File rules2Dir = rules2Dirs.get(0);
+        String actualRules = Files.readString(rules2Dir.listFiles()[0].toPath());
+
+        assertEquals(expectedRules, actualRules);
+    }
+
+    @Test
+    @DisplayName("should export products")
+    void shouldExportProducts() {
+        List<File> productsDir = FileUtil.getSubdirectories("products", export);
+        assertEquals(1, productsDir.size());
+
+        File[] productFiles = productsDir.get(0).listFiles((file, str) -> {
+            return str.toLowerCase().contains(".json");
+        });
+
+        assertEquals(exportFactory.getProductIdToProduct().size(), productFiles.length);
+    }
+
+    @Test
+    @OnlyInHosted
+    @DisplayName("should export products and certs")
+    void shouldExportProductsAndCerts() {
+        List<File> productsDir = FileUtil.getSubdirectories("products", export);
+        assertEquals(1, productsDir.size());
+
+        File[] productFiles = productsDir.get(0).listFiles((file, str) -> {
+            return str.toLowerCase().contains(".json");
+        });
+
+        assertEquals(exportFactory.getProductIdToProduct().size(), productFiles.length);
+
+        int expectedNumberOfCerts = 0;
+        for (File productFile : productFiles) {
+            String expectedName = productFile.getName().replace(".json", "");
+
+            System.out.println("Looking for: " + expectedName);
+
+            ProductDTO product = exportFactory.getProductIdToProduct().get(expectedName);
+            assertNotNull(product);
+
+            // Count numeric ids
+            try {
+                Integer.parseInt(expectedName);
+                expectedNumberOfCerts++;
+            }
+            catch (NumberFormatException e) {
+                // ExpectedName is not a numeric id
+            }
+        }
+
+        File[] certFiles = productsDir.get(0).listFiles((file, str) -> {
+            return str.toLowerCase().contains(".pem");
+        });
+        assertEquals(expectedNumberOfCerts, certFiles.length);
+    }
+
+    @Test
+    @DisplayName("should allow manifest creation with read only user")
+    void shouldAllowManifestCreationWithReadOnlyUser() throws Exception {
+        exportFactory.createExportWithReadOnlyUser();
```

Comment:
We should probably verify the created export exists and has the appropriate properties. There are no assertions at the factory level, so this test only verifies that this particular call completes without an exception-level error.

Really the entirety of that util method should be moved to this test, and then have assertions sprinkled about.

Reply:
I forgot to add a comment on this test because I was confused by the original test. The original Ruby test only had the factory creating the read only export and so I wasn't sure if the test was just making sure that the export could be created without error or not. I moved over the logic from createExportWithReadOnlyUser() into the test and then added assertions on the export!